src/package/mod.rs

```diff
@@ -517,15 +530,86 @@ impl Package {
         Ok(package)
     }
 
-    pub fn from_fpm_doc(fpm_doc: &ftd::p2::Document) -> fpm::Result<Package> {
+    pub fn from_fpm_doc(
+        root: &camino::Utf8Path,
+        fpm_doc: &ftd::p2::Document,
+    ) -> fpm::Result<Package> {
         let temp_package: Option<PackageTemp> = fpm_doc.get("fpm#package")?;
 
-        match temp_package {
-            Some(v) => Ok(v.into_package()),
-            None => Err(fpm::Error::PackageError {
-                message: "FPM.ftd does not contain package definition".to_string(),
-            }),
+        let mut package = match temp_package {
+            Some(v) => v.into_package(),
+            None => {
+                return Err(fpm::Error::PackageError {
+                    message: "FPM.ftd does not contain package definition".to_string(),
+                })
+            }
+        };
+
+        // reading dependencies
+        let mut deps = {
+            let temp_deps: Vec<fpm::package::dependency::DependencyTemp> =
+                fpm_doc.get("fpm#dependency")?;
+            temp_deps
+                .into_iter()
+                .map(|v| v.into_dependency())
+                .collect::<Vec<fpm::Result<fpm::Dependency>>>()
+                .into_iter()
+                .collect::<fpm::Result<Vec<fpm::Dependency>>>()?
+        };
+
+        if package.name != fpm::FPM_UI_INTERFACE
+            && !deps
+                .iter()
+                .any(|dep| dep.implements.contains(&fpm::FPM_UI_INTERFACE.to_string()))
+        {
+            deps.push(fpm::Dependency {
+                package: fpm::Package::new(fpm::FPM_UI_INTERFACE),
+                version: None,
+                notes: None,
+                alias: None,
+                implements: Vec::new(),
+                endpoint: None,
+                mountpoint: None,
+            });
+        };
+        // setting dependencies
+        package.dependencies = deps;
+        package.fpm_path = Some(root.join("FPM.ftd"));
+
+        package.auto_import = fpm_doc
+            .get::<Vec<String>>("fpm#auto-import")?
+            .iter()
+            .map(|f| fpm::AutoImport::from_string(f.as_str()))
+            .collect();
+
+        package.ignored_paths = fpm_doc.get::<Vec<String>>("fpm#ignore")?;
+        package.fonts = fpm_doc.get("fpm#font")?;
+        package.sitemap_temp = fpm_doc.get("fpm#sitemap")?;
+        package.dynamic_urls_temp = fpm_doc.get("fpm#dynamic-urls")?;
+
+        // TODO: resolve group dependent packages, there may be imported group from foreign package
+        // TODO: We need to make sure to resolve that package as well before moving ahead
+        // TODO: Because in `UserGroup::get_identities` we have to resolve identities of a group
```

Comment:
Style comment, this is not three TODOs, just one multi line TODO. So do this:

```suggestion
        // TODO: resolve group dependent packages, there may be imported group from foreign package
        //       We need to make sure to resolve that package as well before moving ahead
        //       Because in `UserGroup::get_identities` we have to resolve identities of a group
```